City-of-Helsinki/drupal-hdbt
# Uhf 2836 service link open in new window

# [UHF-2836](https://helsinkisolutionoffice.atlassian.net/browse/UHF-2836) Open service link in new window on SOTE

## What was done

* A new template was added to drupal-helfi-sote so that SOTE can have it behave differently from rest of the HDBT
* These link buttons had a focus problem as they were set to `float:left` and parent to `overflow:hidden` to contain the float. This was fixed. (This was also reported in this weeks accessibility report)
* Footer had a similar focus problem, was fixed. (This was not reported as a problem yet)
* Bunch of new issues were opened due to problems found with buttons.
* An accessibility audit was done on the approach on how to implement this. 
  * Button should contain both "external link" and "opens in new tab" explanations for screen readers
  * Above button there should be a text explaining to sighted users that the button opens links in new tab
* A bug was fixed in the service box mobile, so that the "Tags" do not break layout, but switch to render below each other.

![image](https://user-images.githubusercontent.com/1191667/144421108-7f0e6217-af18-4343-bfa5-383b1c5711e6.png)

![image](https://user-images.githubusercontent.com/1191667/144421135-3c3d9bad-9d48-4035-95d3-4857ca882a80.png)


## How to test

* Get the [correct branch](https://github.com/City-of-Helsinki/drupal-helfi-sote/pull/241) from drupal-helfi-sote
* Get this branch
* Open a [service page](https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/terveydenhoito/terveysasemat/ajokorttitodistus)
* Check the link, it should have `target="_blank"`
* Check that the link works as planned
* Read the new `link-button.html.twig` template, check that it's a sane approach to link-style buttons
* Check focus style both here and in footer, they should not be hidden with `overflow:hidden` anymore.
* Check that the visible and non-visible help texts are translated properly in all languages.

See also: https://github.com/City-of-Helsinki/drupal-helfi-sote/pull/241
